hyperspy/models/edsmodel.py

```diff
@@ -0,0 +1,881 @@
+
+        Parameters
+        ----------
+        xray_lines: list of str or 'all'
+            The Xray lines. If 'all', fit all lines
+        bounds: float
+            Bound the height of the peak to a fraction of
+            its height
+        """
+        def free_twin():
+            component.A.twin = None
+            component.A.free = True
+            if component.A.value - bound * component.A.value <= 0:
+                component.A.bmin = 1e-10
+                # print 'negative twin!'
+            else:
+                component.A.bmin = component.A.value - \
+                    bound * component.A.value
+            component.A.bmax = component.A.value + \
+                bound * component.A.value
+            component.A.ext_force_positive = True
+        xray_families = [
+            utils_eds._get_xray_lines_family(line) for line in xray_lines]
+        for component in self:
+            if component.isbackground is False:
+                if xray_lines == 'all':
+                    free_twin()
+                elif utils_eds._get_xray_lines_family(
+                        component.name) in xray_families:
+                    free_twin()
+
+    def fix_sub_xray_lines_weight(self, xray_lines='all'):
+        """
+        Fix the weight of a sub X-ray lines to the main X-ray lines
+
+        Establish the twin on the height of sub-Xray lines (non alpha)
+        """
+        def fix_twin():
+            component.A.bmin = 0.0
+            component.A.bmax = None
+            element, line = utils_eds._get_element_and_line(component.name)
+            for li in elements_db[element]['Atomic_properties']['Xray_lines']:
+                if line[0] in li and line != li:
+                    xray_sub = element + '_' + li
+                    component_sub = self[xray_sub]
+                    component_sub.A.bmin = 1e-10
+                    component_sub.A.bmax = None
+                    weight_line = component_sub.A.value / component.A.value
+                    component_sub.A.twin_function = _get_weight(
+                        element, li, weight_line)
+                    component_sub.A.twin_inverse_function = _get_iweight(
+                        element, li, weight_line)
+                    component_sub.A.twin = component.A
+        for component in self.xray_lines:
+            if xray_lines == 'all' or component.name in xray_lines:
+                fix_twin()
+        self.fetch_stored_values()
+
+    def free_xray_lines_energy(self, xray_lines='all', bound=0.001):
+        """
+        Free the X-ray line energy (shift or centre of the Gaussian)
+
+        Parameters
+        ----------
+        xray_lines: list of str or 'all'
+            The Xray lines. If 'all', fit all lines
+        bound: float
+            the bound around the actual energy, in keV or eV
+        """
+
+        for component in self:
+            if component.isbackground is False:
+                if xray_lines == 'all':
```

Comment:
combine the two `if`s:

`if xray_lines == 'all' or component.name in xray_lines:`